test/io-tests.sh

```diff
@@ -88,8 +88,7 @@ readSecretFromFile(){
   while pgrStarted && test "$( rootStatus )" -ne 200
   do
     # wait for the server to start
-    sleep 0.1 \
-    || sleep 1 # fallback: subsecond sleep is not standard and may fail
```

Comment:
@monacoremo I removed the non-standard fallback for sleep - because `nix-shell` does guarantee we're using coreutils' sleep... ehhh this is more of a question. Does it? Or is that taken from the host?

Reply:
That's from before my time here, no idea where this would ever be an issue. LGTM :-) 

Overall those tests are stretching bash to it's limits IMO, it would be a bit easier to meintain as a Python script for example. But it's not too painful as is I guess

Reply:
I agree: Having a proper unit test framework around it, would help. I had a brief look at things to test arbitrary executables. I found DejaGNU and quickly turned away again. But I also found [bats-core](https://github.com/bats-core/bats-core). This is essentially unit testing for bash. The bats syntax is even supported by shellcheck, so this would not block adding linting as well.

The alternative would be to add a python based unit test. Of course this would be more to do for a rewrite. How much upside do you see here @monacoremo? Any experience with python's unit test frameworks?

Reply:
Py.test is super straight forward - just define a few functions in a file (e.g. ljke this https://github.com/monacoremo/daydream/blob/master/tests/test_api_accounts.py) and run.

Happy to take a stab at that

Reply:
I the key question here is: How easy is it to set up the `postgrest` process running in the background, and then clean up this process afterwards? The request side seems super simple.

Probably a helper `postgrest(configPath)` to be used in a `with` block?

```
with postgrest('./configs/simple.config'):
  # run my test
```

Ah - that could be very good!

> Happy to take a stab at that

Yes, please go ahead!